src/rules/heading-blank-lines.ts

```diff
@@ -35,6 +35,9 @@ export default class HeadingBlankLines extends RuleBuilder<HeadingBlankLinesOpti
     text = text.replace(/^\n+(#+\s.*)/, '$1'); // remove blank lines before first heading
     text = text.replace(/(#+\s.*)\n+$/, '$1'); // remove blank lines after last heading
 
+    // blank line between heading and table, see https://github.com/platers/obsidian-linter/issues/751
+    text = text.replace(/(^#+\s.*)(\n\|)/gm, '$1\n$2');
+
```

Comment:
Is there a reason to add this specifically to heading blank lines? I am not following why we are trying to account for tables here if the rule says to remove blank lines after headings. That is expressly being requested? This will affect scenarios where blank lines around tables is not enabled.

Reply:
Ohhhh I see. I just found that there is a [empty line around table rule](https://github.com/platers/obsidian-linter/blob/master/docs/rules.md#empty-line-around-tables). Unfortunately, according to the debug trace, it runs **before** this `heading blank lines` rule. After it adds empty line around table, it gets removed because of this rule.

Probably the `heading-blank-lines` rule should be prioritized? This can simply fix this issue. But for others who don't know there is such a rule like `empty line around table`, maybe add a option (or reminder) to enable `empty line around table`rule?